src/main.cpp

```diff
@@ -4386,7 +4387,15 @@ bool ActivateBestChain(CValidationState& state, const CChainParams& chainparams,
     CBlockIndex *pindexMostWork = NULL;
     CBlockIndex *pindexNewTip = NULL;
     do {
-        boost::this_thread::interruption_point();
+        // Sleep briefly to allow other threads a chance at grabbing cs_main if
+        // we are connecting a long chain of blocks and would otherwise hold the
+        // lock almost continuously. As of 2023-02-03 the Zcash mainnet chain is
+        // around height 1,972,000; this sleep adds ~6.6 minutes to the time
+        // required to activate the best chain from genesis.
+        //
+        // This is defined to be an interruption point.
+        // <https://www.boost.org/doc/libs/1_70_0/doc/html/thread/thread_management.html#interruption_points>
```

Comment:
```suggestion
        // <https://www.boost.org/doc/libs/1_81_0/doc/html/thread/thread_management.html#interruption_points>
```
1.81.0 is the version of boost we [currently depend on](https://github.com/zcash/zcash/blob/feec543abf274d2bb7620720fc8f6833283571b7/depends/packages/boost.mk#L2).